setup/src/Magento/Setup/Controller/DatabaseCheck.php

```diff
@@ -37,12 +38,43 @@ public function indexAction()
         try {
             $params = Json::decode($this->getRequest()->getContent(), Json::TYPE_ARRAY);
             $password = isset($params['password']) ? $params['password'] : '';
-            $this->dbValidator->checkDatabaseConnection($params['name'], $params['host'], $params['user'], $password);
+            $driverOptions = [];
+            if ($this->isDriverOptionsGiven($params)) {
+                if (empty($params['driverOptionsSslVerify'])) {
+                    $params['driverOptionsSslVerify'] = 0;
+                }
+                $driverOptions = [
+                    ConfigOptionsListConstants::KEY_MYSQL_SSL_KEY => $params['driverOptionsSslKey'],
+                    ConfigOptionsListConstants::KEY_MYSQL_SSL_CERT => $params['driverOptionsSslCert'],
+                    ConfigOptionsListConstants::KEY_MYSQL_SSL_CA => $params['driverOptionsSslCa'],
+                    ConfigOptionsListConstants::KEY_MYSQL_SSL_VERIFY => (int) $params['driverOptionsSslVerify'],
+                ];
+            }
+            $this->dbValidator->checkDatabaseConnectionWithDriverOptions(
+                $params['name'],
+                $params['host'],
+                $params['user'],
+                $password,
+                $driverOptions
+            );
             $tablePrefix = isset($params['tablePrefix']) ? $params['tablePrefix'] : '';
             $this->dbValidator->checkDatabaseTablePrefix($tablePrefix);
             return new JsonModel(['success' => true]);
         } catch (\Exception $e) {
             return new JsonModel(['success' => false, 'error' => $e->getMessage()]);
         }
     }
+
+    /**
```

Comment:
Please add a description according to https://devdocs.magento.com/guides/v2.3/coding-standards/docblock-standard-general.html#inheritdoc